src/core/components/icon/icon.tsx

```diff
@@ -48,11 +51,17 @@ const responsiveSizeClassNames: ResponsiveAttributeClassMap<IconSize> = {
 };
 
 export const Icon: React.FC<IIconProps> = (props) => {
-    const { icon, size = 'md', className, responsiveSize = {}, ...otherProps } = props;
+    const { icon, size = 'md', className, responsiveSize, ...otherProps } = props;
 
     const IconComponent = iconList[icon];
 
     const sizeClassNames = responsiveUtils.generateClassNames(size, responsiveSize, responsiveSizeClassNames);
 
-    return <IconComponent className={classNames('shrink-0', sizeClassNames, className)} {...otherProps} />;
+    return (
+        <IconComponent
+            className={classNames('shrink-0', sizeClassNames, className)}
+            data-testid={icon}
```

Comment:
The icons already have a testid on the SVG per file I believe? It's the name of the icon.

Reply:
Yess, it already works internally on the `ods` library because we transform svgs into React components with test-ids through the `/core/test/svgTransform.js` file. The `data-testid` is needed for applications using the `ods` library (app-next) because the svgs are bundled into the library and cannot be transformed as we already do on the library itself. Good point though 👌🏻